packages/core/src/browser/widget-manager.ts

```diff
@@ -106,8 +106,12 @@ export class WidgetManager {
     protected async doGetWidget<T extends Widget>(key: string): Promise<T | undefined> {
         const existingWidgetPromise = this.widgetPromises.get(key);
         if (existingWidgetPromise) {
-            const existingWidget = await existingWidgetPromise;
-            return existingWidget as T;
+            try {
+                return await existingWidgetPromise as T;
+            } catch {
```

Comment:
we do not want to re-throw, right?

Reply:
I did not want to. Do you think it would make sense? The purpose of the change is only to avoid the caching.

Reply:
Only if call site would have handled that error, but I haven't found any error handling. 
👍

Reply:
Would it swallow a reason why a widget cannot be created?

Reply:
Let's log that.

Reply:
Done.